apiserver/routers.go

```diff
@@ -41,5 +42,34 @@ func (api *APIServer) NewRouter(ctx context.Context) *mux.Router {
 		Name("GetDistroSigImageConfig").
 		HandlerFunc(api.GetDistroSigImageConfig)
 
+	router.Methods("GET").Path("/healthz").Name("healthz").HandlerFunc(healthz)
+
+	// global timeout and panic handlers.
+	router.Use(timeoutHandler(), recoveryHandler())
+
 	return router
 }
+
+func healthz(w http.ResponseWriter, r *http.Request) {
+	handleOK(w, r)
+}
+
+func setHeaders(w http.ResponseWriter) {
+	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+}
+
+func handleOK(w http.ResponseWriter, r *http.Request) {
+	setHeaders(w)
+	w.WriteHeader(http.StatusOK)
+}
+
+func recoveryHandler() mux.MiddlewareFunc {
+	return handlers.RecoveryHandler(handlers.PrintRecoveryStack(true))
```

Comment:
this captures panics and logs them, which ironically should be standard go http behavior, but because we used goroutines we escaped the panic handler in the location of the panic.

Reply:
correction: the default is recovery but does not include the panic stack from the handler apparently.